msg.h

```diff
@@ -37,6 +38,10 @@ typedef struct _msg_t {
 	u32 csum;
 	u32 type;
 	u8  data[MSG_MAXLEN];
+	union {
+		int (*read)(unsigned int, u8 *, u16, u16);
+		int (*write)(unsigned int, const u8 *, u16);
+	};
```

Comment:
Should this be an union? It means that both pointers of different types point to the same function implicitly. Might cause some headache in the future

Reply:
Only one of the callback is used by` msg_t ` and it is controlled by phoenixd  so I think that union in this case is used correctly.